src/java.desktop/share/classes/javax/swing/JMenu.java

```diff
@@ -1255,6 +1255,18 @@ public void applyComponentOrientation(ComponentOrientation o) {
         }
     }
 
+    /**
+     * Sets the orientation for this menu and the associated popup component
+     * determined by the <code>ComponentOrientation</code> argument.
+     *
+     * @param  o one of the following values:
+     * <ul>
+     * <li>java.awt.ComponentOrientation.LEFT_TO_RIGHT
+     * <li>java.awt.ComponentOrientation.RIGHT_TO_LEFT
+     * <li>java.awt.ComponentOrientation.UNKNOWN
+     * </ul>
+     * @see java.awt.ComponentOrientation
+     */
```

Comment:
Shall it just use `{@inheritDoc}`?
The javadoc generated in 7 and 8 does exactly this.

However, the new text explicitly mentions that this method changes not only orientation of this component but also of the associated popup menu.

I do not think the description of the parameter has to list all the values of `ComponentOrientation`, these are the only publicly available values any way. If a value is added in the future, the javadoc will become incorrect.

Reply:
It is done similarly in JScrollPane which I used here...maybe there also it's an oversight then..
I will update to use @inheritdoc in this spec ...

Reply:
Yes, I see… Yet I think listing values of `ComponentOrientation` is redundant.

However, looking at JScrollPane, its javadoc specifies why it's overridden: _Sets the orientation for the vertical and horizontal scrollbars_. In the case of `JMenu`, it also changes the orientation of *popup menu* inside JMenu. Taking into account this additional detail, your initial text could have been better.